data/data.go

```diff
@@ -0,0 +1,422 @@
+// *sql.Tx implement it.
+type Execer interface {
+	// Exec executes a query without returning any rows. The args are for any
+	// placeholder parameters in the query.
+	Exec(query string, args ...interface{}) (sql.Result, error)
+}
+
+// NullTime represents a time.Time that may be null. NullTime implements the
+// sql.scanner interface so it can be used as a scan destination, similar to
+// sql.NullString.
+type NullTime struct {
+	Time  time.Time
+	Valid bool // Vaid is true if Time is not NULL
+}
+
+// Scan implements the sql.scanner interface.
+func (n *NullTime) Scan(value interface{}) error {
+	if value == nil {
+		*n = NullTime{}
+		return nil
+	}
+	n.Valid = true
+	if t, ok := value.(time.Time); !ok {
+		return fmt.Errorf("can't save %T as time.Time")
+	} else {
+		n.Time = t
+	}
+	return nil
+}
+
+// scanner is an interface to retrieve values from an sql row. Both *sql.Row
+// and *sql.Rows implement it.
+type scanner interface {
+	// Scan copies the columns from the matched row into the values pointed at
+	// by dest. See the documentation on Rows.Scan for details. If more than
+	// one row matches the query, Scan uses the first row and discards the
+	// rest. If no row matches the query, Scan returns ErrNoRows.
+	Scan(dest ...interface{}) error
+}
+
+// row is an abstraction over the data elements to share code.
+type row interface {
+	// selectFragment returns an SQL query fragment of the form "SELECT field1,
+	// field2,... " with all columns for this type.
+	selectFragment() string
+
+	// scanFrom calls the scanners Scan method to retrieve the fields in the
+	// order output by selectFragment.
+	scanFrom(scanner) error
+}
+
+// queryRow is a convenience function to get a single data row.
+func queryRow(q Querier, r row, constrict string, args ...interface{}) error {
+	row := q.QueryRow(r.selectFragment()+constrict, args...)
+	return r.scanFrom(row)
+}
+
+// TerminIterator is an iterator over a subset of the termine table.
+type TerminIterator struct {
+	rows *sql.Rows
+	err  error
+	t    *Termin
+}
+
+// Next returns whether there is another element.
+func (it *TerminIterator) Next() bool {
+	if it.err != nil {
+		return false
+	}
+	if !it.rows.Next() {
+		it.err = it.rows.Err()
+		return false
+	}
+	it.t = new(Termin)
+	it.err = it.t.scanFrom(it.rows)
+	return it.err == nil
+}
+
+// Val returns the current row. Requires Next() to be true.
+func (it *TerminIterator) Val() *Termin {
+	return it.t
+}
+
+// Err returns the last error that occured when reading. It must be called
+// after being done with the iterator.
+func (it *TerminIterator) Err() error {
+	err := it.rows.Close()
+	if it.err == nil {
+		it.err = err
+	}
+	return it.err
+}
+
+// One is a convenience method for queries that expect exactly one result. It
+// returns an error if there is not exactly one result. Must be the only method
+// being called.
+func (it *TerminIterator) One() (*Termin, error) {
```

Comment:
They serve different purposes (so far, One doesn't serve any, though). One is useful for "I am searching for a specific meeting", whereas First is useful for e.g. "I want the next meeting". In the former case, using First would potentially hide bugs in the query or unexpected data.

I'm not married to it, though. If you want, I can remove it until the need is demonstrated.